packages/sdk/README.md

```diff
@@ -16,89 +16,264 @@ Get the current api version of the Mainframe client.
 
 **returns:** `Promise<string>`
 
-## Blockchain
+## Contacts
 
-A set of calls that enable interaction with the Ethereum blockchain.
+### contacts.selectContacts()
 
-### blockchain.getContractEvents()
+Will present the user with a list of contacts to select from and returns an Array containing data for the selected contacts.
 
-Read events from the blockchain.
+**returns:** Promise<Array<[Contact](#contact)>>
+
+**Example:**
+
+```
+const contacts = await sdk.contacts.selectContacts()
+```
+
+### contacts.selectContact()
+
+Will allow the user to only select a single contact from their list of contacts and return a data object for that contact. If no contact is selected it will return undefined.
+
+**returns:** Promise<?[Contact](#contact)>
+
+**Example:**
+
+```
+const contact = await sdk.contacts.selectContact()
+```
+
+### contacts.getDataForContacts(contactIDs: Array<string\>)
+
+Allows you to fetch data for contacts your app has previously been approved to read data from. It's possible some contacts could return no data in the case where the user has deleted a contact or revoked permission.
+
+**returns:** Promise<Array<[ExistingContact](#existing-contact)>>
+
+**Example:**
+
+```
+const contacts = await sdk.contacts.getDataForContacts(['ue64gf93b'])
+```
+
+### contacts.getDataForContact(contactID: string)
+
+A call to fetch data for a single contact your app has previously been approved to read data from.
+
+**returns:** Promise<?[Contact](#contact)>
```

Comment:
Shouldn't this return `ExistingContact` rather than `Contact`?

Reply:
The idea is it only returns a full Contact and null if contact has been removed or revoked, the ExistingContact type is mainly to support fetching a list, to make it obvious which contacts from your list of ID's aren't accessible any longer.

Reply:
👍 makes sense, thanks!